dvc/cache/base.py

```diff
@@ -247,6 +248,116 @@ def _save_file(self, path_info, tree, hash_info, save_link=True, **kwargs):
 
         self.tree.state.save(cache_info, hash_info)
 
+    def _transfer_file_as_whole(self, from_tree, from_info):
+        from dvc.utils import tmp_fname
+
+        # When we can't use the chunked upload, we have to first download
+        # and then calculate the hash as if it were a local file and then
+        # upload it.
+        local_tree = self.repo.cache.local.tree
+        local_info = local_tree.path_info / tmp_fname()
+
+        from_tree.download(from_info, local_info)
```

Comment:
This might not fit on the local filesystem :slightly_frowning_face: We could download a chunk and upload a chunk though and then assemble them. Or could just forbid transfering files as a whole for now.